authorization.md

```diff
@@ -39,14 +40,35 @@ consider the [OAuth 2.0 Dynamic Client Registration
 Protocol](https://tools.ietf.org/html/draft-ietf-oauth-dyn-reg) for an
 out-of-the-box solution.
 
-No matter how a backend service registers with an EHR's authorization service,
-at registration time **every SMART backend service must**:
+No matter how a backend service registers with an EHR's authorization service, a 
+backend service should communicate its **public key** to the SMART EHR using a
+[JSON Web Key Set (JWKS)](https://tools.ietf.org/html/rfc7517) by one of the
+following techniques:
 
-* Register a fixed "issuer URL" with the EHR
-* Register a public RSA key with the EHR (for RSA SHA-256 signatures)
+  1. JWKS URL (preferred). This URL communicates the TLS-protected endpoint where the service's
+  JSON Web Key can be found. When provided, this URL will match the `jku` header
+  parameter in the service's Authorization JWTs. An advantage of this approach is that
+  it allows a client to rotate its own keys by updating the hosted content at the JWKS URL.
+  2. JWKS directly (allowed, not preferred). If a backend service cannot host a JWKS at a
+  TLS-protected URL, it may supply a JWKS directly to the EHR at registration time. A limitation
+  of this approach is that it does not enable the client to rotate its keys in-band.
+  
+It is recommended that EHRs should be capable of validating signatures using `RS384` and `ES384`; and that
+backend services be capable of generating signatures using one of these two
+algorithms. Over time, we expect recommended algorithms to evolve, so while
+while this specification recommends algorithms for interoperability, it does
+not mandate any algorithm.
 
-Upon registration, the server assigns a `client_id`, which 
-the client uses when obtaining an access token.
+No matter how a JWKS is communicated to the EHR, each key in the JWKS *must be* an asymmetric
+key whose content is conveyed using "bare key" properties (i.e., direct base64 encoding of 
+key material as integer values). This means that:
+
+* For RSA public keys, each MUST include `n` and `e` values (modulus and exponent) 
```

Comment:
Should we update the worked example below to match this - it's 404ing on the public key, but in any case the one included in the original backend services docs on isn't a JWKS?

Reply:
Yes, we should update examples. I will do this before merging.

Reply:
Update the example and added `authorization-example-jwks-and-signatures.ipynb` and `authorization-example-jwks.json`.